src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -85,16 +85,37 @@ impl FromStr for MapsEntry {
     // e.g.: "ffffffffff600000-ffffffffff601000 --xp 00000000 00:00 0                  [vsyscall]"
     // e.g.: "7f5985f46000-7f5985f48000 rw-p 00039000 103:06 76021795                  /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2"
     // e.g.: "35b1a21000-35b1a22000 rw-p 00000000 00:00 0"
+    //
+    // Note that paths may contain spaces, so we can't use `str::split` for parsing (until
+    // Split::remainder is stabalized #77998).
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let mut parts = s
-            .split(' ') // space-separated fields
-            .filter(|s| s.len() > 0); // multiple spaces implies empty strings that need to be skipped.
-        let range_str = parts.next().ok_or("Couldn't find address")?;
-        let perms_str = parts.next().ok_or("Couldn't find permissions")?;
-        let offset_str = parts.next().ok_or("Couldn't find offset")?;
-        let dev_str = parts.next().ok_or("Couldn't find dev")?;
-        let inode_str = parts.next().ok_or("Couldn't find inode")?;
-        let pathname_str = parts.next().unwrap_or(""); // pathname may be omitted.
+        let (range_str, s) = s.trim().split_once(' ').unwrap_or((s, ""));
```

Comment:
No need to trim until you reached pathname, yes?

Reply:
Without the trim multiple spaces in a row would be considered individual split targets rather than a single one, right? So for example `a<space><space>b` would split as `["a", "", "b"]`.

Reply:
From new test, but appended trailing spaces to path `"b7c79000-b7e02000 r--p 00000000 08:01 60662705   /executable/path/with some spaces       "`, this will be broken.

Reply:
Replaced `trim` with `trim_start` where appropriate!